### PR TITLE
Setup Redux Review

### PR DESCRIPTION
**Pull request by akezeth**
- Added Redux Toolkit.
- Created a directory that will contain all Redux logic (/src/redux).
- Configured a Redux store (/src/redux/store.js).
- Defined a slice of state for books.
- Included an array of books (initial state: empty array).
- Included a reducer that adds a book.
- Included a reducer that removes a book.
- Defined a slice of state for categories.
- Included an array of categories (initial state: empty array)
- Included a reducer that checks the status and always returns "Under construction" (the initial state should check to that string).

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export const booksSlice = createSlice({
+  name: 'book',
+  initialState: [],
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.filter((book) => book.id !== action.payload.id);
```
Comment (muneebulrehman) [inline src/redux/books/booksSlice.js]: The `filter()` method here does not modify the existing array but returns a new filtered array. Therefore, you either need to assign the filtered array back to the state for the changes to take effect or put a `return` statement before filtering. Kindly update your code accordingly.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @akezeth , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Proper redux toolkit setup.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by akezeth** [comment]
- Adjust filter array statement in 'removebook' reducer.

**Review by sumairq — APPROVED** [review]
### Hi @akezeth , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**